### PR TITLE
Add mapbox

### PR DESCRIPTION
**Pull request by blamb888**
RpgSeznamka - groups map
RpgSeznamka - groups search zoom
So I went ahead and set up the basics for mapbox. The stuff for markers is there but not being used atm. I also added some routes for groups and basic view setup for the maps. I put autocomplete on the search control for the map as well. There is also some basic scss for the mapbox popups if we decide to use them in the future.

### CODE REVIEW

**Comment by blamb888** [comment]
Also - yarn add @mapbox/mapbox-gl-geocoder - for map search controls

**Comment by blamb888** [comment]
And - yarn add places.js - for autocomplete in the search bar

File: app/controllers/groups_controller.rb
Comment (Naokimi) [inline app/controllers/groups_controller.rb]: we want to keep the whole app behind sign_in wall, so don't skip authentication

File: app/controllers/groups_controller.rb
Comment (Naokimi) [inline app/controllers/groups_controller.rb:4]: let's have `Group.where` preference: preference (it's a bit more complicated than that, try to look into it and if you can't figure it out let me know)
the preference comes from params - if there are no preferences in params let's use the preferences that the current_user has selected

File: app/views/groups/_info_window.html.erb
Comment (Naokimi) [inline app/views/groups/_info_window.html.erb]: use html.slim template instead of html.erb

File: app/views/groups/index.html.erb
Comment (Naokimi) [inline app/views/groups/index.html.erb]: use html.slim template instead of html.erb

File: config/routes.rb
Comment (Naokimi) [inline config/routes.rb]: don't add all the resources if the controller actions haven't been added yet

File: db/migrate/20210324091636_add_coordinates_to_groups.rb
Comment (Naokimi) [inline db/migrate/20210324091636_add_coordinates_to_groups.rb:3]: do we need this migration if we are searching mapbox by city rather than exact address?
I'm genuinely asking here

**Review by Naokimi — CHANGES_REQUESTED** [review]
please check that your code is following rubocop
you have some extra spaces in the groups_controller

also, you are missing tests and Pundit for the controller

**Comment by Naokimi** [comment]
addendum:
thinking back your scope was only adding mapbox, so I guess we can skip for now the requests on the controller and have that as a different ticket
but please fix the other points mentioned in the review

File: db/migrate/20210324091636_add_coordinates_to_groups.rb
Comment (blamb888) [reply db/migrate/20210324091636_add_coordinates_to_groups.rb:3]: I'm honestly not 100% sure. From following the Le Wagon slides I'm like 50/50 that Mapbox might be using it to geolocate. Again, not 100% on that, though.

File: app/views/groups/_info_window.html.erb
Comment (blamb888) [reply app/views/groups/_info_window.html.erb]: I've never used that before so I might make some syntax errors out the gate. It looks nice though. I'll go over the docs a bit today and make the changes.

File: app/views/groups/index.html.erb
Comment (blamb888) [reply app/views/groups/index.html.erb]: I've never used that before so I might make some syntax errors out the gate. It looks nice though. I'll go over the docs a bit today and make the changes.

File: db/migrate/20210324091636_add_coordinates_to_groups.rb
Comment (Naokimi) [reply db/migrate/20210324091636_add_coordinates_to_groups.rb:3]: ok, let's keep them in for now

File: app/controllers/groups_controller.rb
Comment (blamb888) [reply app/controllers/groups_controller.rb]: I'm having trouble signing up. Is there some validation setting on devise or pundit rn preventing new sign ups? 
image

File: config/routes.rb
Comment (blamb888) [reply config/routes.rb]: kk nested and changed to just index rn
image

**Comment by blamb888** [comment]
> addendum:
> thinking back your scope was only adding mapbox, so I guess we can skip for now the requests on the controller and have that as a different ticket
> but please fix the other points mentioned in the review

Okay cool. So I'll let you know when I update the html.erb file to html.slim and I'll leave the controller/scope. Which is good because I think I need to go rewatch the pundit lecture otherwise, lol

File: config/routes.rb
Comment (Naokimi) [reply config/routes.rb]: if you do like this it means the url will be /users/:user_id/groups, meaning we'll get a list of all groups _belonging to a user_
but that's not what we want, what we want is a list of all groups, period (plus the filtering mentioned above)

File: app/controllers/groups_controller.rb
Comment (Naokimi) [reply app/controllers/groups_controller.rb]: best to check logs in rails server, what is telling you

File: app/controllers/groups_controller.rb
Comment (Naokimi) [inline app/controllers/groups_controller.rb]: no need to write tests but please remove this line

File: app/views/groups/_info_window.html.slim
Comment (Naokimi) [inline app/views/groups/_info_window.html.slim:1]: where does this appear? didn't notice on the shared screenshot

File: app/views/groups/index.html.slim
Comment (Naokimi) [inline app/views/groups/index.html.slim:3]: are the markers appearing?

File: app/models/group.rb
Comment (Naokimi) [inline app/models/group.rb]: you might want to wait until we merge https://github.com/Naokimi/rpg-seznamka/pull/12 or implement the changes here, because otherwise this will break

File: config/routes.rb
Comment (Naokimi) [inline config/routes.rb]: if you do like this it means the url will be /users/:user_id/groups, meaning we'll get a list of all groups _belonging to a user_
but that's not what we want, what we want is a list of all groups, period (plus the filtering mentioned above)

**Review by Naokimi — CHANGES_REQUESTED** [review]

File: config/routes.rb
Comment (blamb888) [reply config/routes.rb]: image
I don't know how to add the filtering. Hope this is what you're looking for.

File: app/controllers/groups_controller.rb
Comment (blamb888) [reply app/controllers/groups_controller.rb]: Done

File: app/views/groups/_info_window.html.slim
Comment (blamb888) [reply app/views/groups/_info_window.html.slim:1]: This would be a popup that activated on-click of the group marker on the mapbox map

File: app/views/groups/index.html.slim
Comment (blamb888) [reply app/views/groups/index.html.slim:3]: Haven't tested but it should. Need some seeds.

File: app/models/group.rb
Comment (blamb888) [reply app/models/group.rb]: image

**Review by Naokimi — APPROVED** [review]

**Pull request merged**
